consumer/src/main/java/com/example/consumer/transaction/TransactionListener.java

```diff
@@ -0,0 +1,31 @@
+package com.example.consumer.transaction;
+
+import com.example.consumer.consumer.KafkaConsumerService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+import static com.example.consumer.consumer.TopicName.TRANSACTIONS;
+
+@Slf4j
+@Component
+public class TransactionListener implements ApplicationListener<ApplicationReadyEvent> {
+
+    @Autowired
+    private KafkaConsumerService<TransactionDto> kafkaConsumerService;
+    @Autowired
+    private TransactionService transactionService;
+
+    @Override
+    public void onApplicationEvent(final ApplicationReadyEvent event) {
+        log.info("---------TransactionListener is working-------------");
+        kafkaConsumerService.subscribeToTopic(
+                TRANSACTIONS.getTopicName(),
+                TransactionDto.class,
+                "transaction-group",
```

Comment:
minor: it's fine for the sake of the exercise, but it's always better to store all the kafka configs (i.e. group.id, bootstrap.servers, topic names, auto.offset.reset, etc) outside the application (e.g. in config files)

Reply:
updated